include/mesh/mesh_tools.h

```diff
@@ -360,13 +360,23 @@ void correct_node_proc_ids(MeshBase &);
  */
 void clear_spline_nodes(MeshBase &);
 
-#ifdef DEBUG
 /**
- * A function for verifying that an element has been cut off
- * from the rest of the mesh
+ * A function for testing whether a mesh's cached is_prepared() setting
+ * is not a false positive.  If the mesh is marked as not prepared, or
+ * if preparing the already-partitioned mesh (without any
+ * repartitioning or renumbering) does not change it, then we return
+ * true.  If the mesh believes it is prepared but prepare_for_use()
+ * would change it, we return false.
  */
-void libmesh_assert_no_links_to_elem(const MeshBase & mesh,
-                                     const Elem * bad_elem);
+bool valid_is_prepared (const MeshBase & mesh);
+
+/**
+ * The following functions, only available in builds with NDEBUG
+ * undefined, are for asserting internal consistency that we hope
+ * should never be broken in opt
+ */
```

Comment:
I think you can use [grouping](https://www.doxygen.nl/manual/grouping.html) if you actually want to get some doxygen for this comment. Otherwise I don't know what doxygen does with this?